src/Promitor.Agents.Scraper/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -236,15 +237,20 @@ private static void AddOpenTelemetryCollectorMetricSink(string collectorUri, str
                     {
                         metricsBuilder.SetResourceBuilder(resourceBuilder)
                                       .AddMeter("Promitor.Scraper.Metrics.AzureMonitor")
-                                      .AddOtlpExporter(options => options.Endpoint = new Uri(collectorUri));
+                                      .AddOtlpExporter(options =>
+                                          {
+                                              options.Endpoint = new Uri(otelConfiguration.CollectorUri);
+                                              options.Protocol = (otelConfiguration.CollectorProtocol.StartsWith("http")) ? OtlpExportProtocol.HttpProtobuf : OtlpExportProtocol.Grpc;
```

Comment:
Let's not assume and use the improved configuration you added below.

Also, let's make sure we add validation

Reply:
I think I did this part right....